### PR TITLE
MAKE-1041: add reverse proxy handler

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: proxy.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	TargetPool        []string
+}
+
+// Validate checks the default configuration of a proxy configuration.
+func (opts *ProxyOptions) Validate() error {
+	if !strings.HasPrefix(opts.RemotePrefix, "/") {
+		opts.RemotePrefix = "/" + opts.RemotePrefix
+	}
+
+	catcher := grip.NewBasicCatcher()
```
Comment (bsamek) [inline proxy.go]: Seems slightly silly to have a catcher for a single error.

File: proxy.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			"version":          r.version,
+			"existing_handler": r.handler != nil,
+		}))
+		return r
+	}
+
+	r.handler = (&httputil.ReverseProxy{
+		ErrorLog: grip.MakeStandardLogger(level.Warning),
+		Director: opts.director,
+	}).ServeHTTP
```
Comment (bsamek) [inline proxy.go]: Are we really not calling ServeHTTP, just adding the function object?

Comment (tychoish) [reply proxy.go]: yeah, r.handler is the handler for the route, the route builder registers function objects or converts handler implementations into handler functions.

File: proxy.go (hunk 1 above)
Comment (tychoish) [reply proxy.go]: yeah, but it makes it easy to add more things.

**Pull request merged**